### DISCUSSION TITLE
Avoiding unforeseen UI lint errors

### DISCUSSION THREAD

**Discussion by leecalcote** [meshery/meshery#1105] [General] [answered]
@kushthedude we're in need of updating our contributor docs to ensure that UI lint errors and failed builds aren't a surprise. What steps do we need to take in context of these from our session last night - https://github.com/layer5io/meshery/runs/737371452?check_suite_focus=true#step:5:205

**Comment by kushthedude** [comment, marked as answer]
I think rather than having frequent failed builds, we should make the UI Lint check as mandatory. We already have make run-ui-lint command but this command wont fix every error for you, now contributors need to be aware if they are going to write poor ui code, the ci will fail.
What we can do is to have a pre-commit so that the contributor can be warned that your code have sucu errors, also we can update the PR template to add a point have you run make run-ui-lint before pushing?

**Reply by kushthedude** [reply]
It has been fixed @leecalcote! We have introduced contributing docs & also have merged eslint with `run-ui-dev` command.

**Reply by leecalcote** [reply]
Looks good. Thank you, @kushthedude.
